### DISCUSSION TITLE
The ability to execute code in the background on Android and iOS

### DISCUSSION THREAD

**Discussion by normalllll** [fzyzcjy/flutter_rust_bridge#2370] [General]
I hope that when the app switches to the background on the mobile device, it can still continue to execute code, such as file downloading.

I have considered creating an `isolate` to call `RustLib.init`, and then download files, but I don't know if it is appropriate.

```dart
void _isolateDownload(_DownloadIsolateConfig args) async {

  final receivePort = ReceivePort();

  args.mainSendPort.send(receivePort.sendPort);

  await RustLib.init();
  ......

  await for (final message in receivePort) {
    if (message is _DownloadTaskArgs) {
      final String url = message.url;
      final String savePath = message.filePath;
      final SendPort taskSendPort = message.taskSendPort;
      ......
      //Calling Rust for file download
    }
  }
}
```

**Comment by fzyzcjy** [comment]
IIRC you need to search for how each platform allows/disallows this. For example, for android, my old memory (can be inaccurate) is to have a `Service` or something like that, otherwise your thread (and process) will be all killed. There may be some Dart/Rust packages that help you with that.

**Comment by normalllll** [comment]
He shouldn't be as heavyweight as `Worker`. The problem I encountered is that if I call Rust directly in Main Isolate, the download task will stop as soon as the App switches to the background (error). What I need is for the user to manually set it to allow background execution, and then the App switches to the background and can still continue to execute Rust code to download files for a short period of time.

**Reply by fzyzcjy** [reply]
Maybe you need to search Android/iOS doc (and maybe Flutter wrapper package doc) for that. IIRC the process kill is not guaranteed not to happen unless you follow some specific operations they require.

Btw, this is not related to flutter_rust_bridge: You can search "how to do it in pure Flutter" and "how to do it in pure Rust" and the results will apply.
